client/src/ui/components/Profile.tsx

```diff
@@ -0,0 +1,218 @@
+import React, { useRef, useState } from "react";
+import Dropdown from "./Dropdown";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faEdit } from "@fortawesome/free-solid-svg-icons";
+import { faStar } from "@fortawesome/free-solid-svg-icons";
+// import ThemeContext from "./ThemeContext";
+
+interface ProfileProps {
+  gameTags: string[];
+  username: string;
+  description: string;
+  profilePicture: string;
+  membership: boolean;
+  auth: boolean;
+}
+
+const Profile: React.FC<ProfileProps> = ({
+  profilePicture,
+  username,
+  description,
+  gameTags,
+  membership,
+  auth,
+}) => {
+  // const authenticated = useContext(Authenticated)
+  // Needs to pass a profile through
+  // let authenticated = true;
+
+  const [editDescription, setEditDescription] = useState(description);
+  const [tempDescription, setTempDescription] = useState(description);
+  const [editProfilePic, setEditProfilePic] = useState(profilePicture);
+  const [tempProfilePic, setTempProfilePic] = useState(profilePicture);
+  const [editGameTags, setEditGameTags] = useState(gameTags);
```

Comment:
These variables named `editX` are a bit confusing - it looks like they're meant to be the actual saved data that matches what's in the database? If so, I would just rename it to remove the `edit` (so `editProfilePic` -> `profilePic` etc.), usually variables starting with a verb like "edit" are functions